### PR TITLE
overlord: prune and abort old changes and tasks

### PR DESCRIPTION
**Pull request by niemeyer**
Initial delay is set to prune after 1 day after a change is ready,
and to abort a change after 7 days of it being non-ready.

### CODE REVIEW

File: overlord/overlord_test.go [hunk 1]
```diff
@@ -214,6 +214,35 @@ func (ovs *overlordSuite) TestEnsureLoopMediatedEnsureBeforeInEnsure(c *C) {
...
+	st.Lock()
+	t1 := st.NewTask("foo", "...")
+	chg1 := st.NewChange("abort", "...")
+	chg1.AddTask(t1)
+	chg2 := st.NewChange("prune", "...")
+	chg2.SetStatus(state.DoneStatus)
+	st.Unlock()
+
+	o.Loop()
+	time.Sleep(50 * time.Millisecond)
```
Comment (pedronis) [inline overlord/overlord_test.go]: this kind of sleep based tests tend to be fragile on slow ci , not a blocker but we will have to revisit this (we have already a flaky one somewhere else under overlord that I need to take a look at)

**Comment by pedronis** [comment]
LGTM, one comment about the test but not a blocker

**Comment by zyga** [comment]
Looks good. I love how ``Prune`` works internally. Nice and clean.

File: overlord/overlord_test.go (hunk 1 above)
Comment (niemeyer) [inline overlord/overlord_test.go]: Good point, thanks. I won't handle it right now to avoid adding more uncertainty, but let's fix this.

**Pull request merged**